### DISCUSSION TITLE
New-BurntToastNotification throws "Access is denied."

### DISCUSSION THREAD

**Discussion by timjdowning** [Windos/BurntToast#98] [General]
I'm new to BurntToast. Just installed 0.71 and running PowerShell as administrator. Please see -WhatIf attached.

Any help appreciated!

Tim

[ps_console.txt](https://github.com/Windos/BurntToast/files/4773853/ps_console.txt)

**Comment by Windos** [comment]
@timjdowning I'm away from PC ATM so can't give exact commands. Can you run $PSVersionTable and also let me know which version of Windows you're on and off it's 32 bit or 64 bit

**Reply by timjdowning** [reply]
See $PSVersionTable below. Using Windows 1909. As discussed, this issue was fixed by running PS in the local user context (not administrator)

> PSVersion                      5.1.18362.752                                                                          
> PSEdition                      Desktop                                                                                
> PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                
> BuildVersion                   10.0.18362.752                                                                         
> CLRVersion                     4.0.30319.42000                                                                        
> WSManStackVersion              3.0                                                                                    
> PSRemotingProtocolVersion      2.3                                                                                    
> SerializationVersion           1.1.0.1          

                                                                      
`

**Comment by timjdowning** [comment]
My use case is:

- to present a 'wash/sanitise your hands' reminder at regular (slightly randomised intervals) while the user is logged on and the computer is unlocked. Timer resets when computer is unlocked (optional)

- to present 2 buttons: "will do it now", and "not this time"

Params for TimerLow (60) and TimerHigh (75) for minimum and maximum time before repeat toast.

Buttons will link to a webhook and pass date/time, env:USERNAME in json format

![2020-06-13 13_04_22-](https://user-images.githubusercontent.com/66856135/84558488-8e197280-ad76-11ea-9567-5c1ba0b1ff9f.png)

Please give me a DM if this is something you could do for us and an approximate cost.

**Reply by Windos** [reply]
Unfortunately I don't have capacity for a full solution, but I can give you the building blocks.

First, as you've found you need the toast to pop as the user. One way to do this is via Scheduled Tasks/Jobs. If you register them for the user but only "when the user is logged in" then you don't need their password.

On this scheduled task you set an ongoing 60 minute trigger and also a random delay up to 15 minutes or so.

Alternatively you can use [Invoke-CommandAs](https://www.powershellgallery.com/packages/Invoke-CommandAs):

```powershell
Invoke-CommandAs -ScriptBlock { # Do Toast here } -AsInteractive 'username'
```

Behind the scenes that's using scheduled tasks too.

Next you'll need a script that fires off to the webhook, I would suggest you make it take a parameter that's either looking for "now" or "later" which will be determined by which button the user presses.

In order to tie the toast and the webhook script together, you'll need a "URI Protocol Handler". I did a demo on this recently and the files for it are sitting in a [PR for the user group](https://github.com/rtpsug/Meetups/pull/12/files) atm.

I'll do a bit of a write up on this part of the process later today...

**Reply by Windos** [reply]
Re the protocol handler:

Step one is some registry settings, in those files they're specified in "ToastPS.reg". Two things about this, it defines the name of the protocol (in the example it's **ToastPS**) and the path to a cmd file (**C:\ToastDemo\ToastPS.cmd**).

You can make those whatever.

The cmd file just calls PowerShell.exe, the thing to change in there is the path to your ps1 file (in your case this will be the file that fires the webhook).

The cmd file passes in arguments to powershell in the form of "Protocol:Arguments" so you'll likely want to strip out the protocol part (in my script you'll see `$Wallpaper = ($args[0] -split ':')[1]`)

Finally in your toast, you want an argument on each of your buttons. Assuming your protocol is still "ToastPS" and your script is expecting "now" or "later" you would make the buttons as such:

```powershell
$NowButton = New-BTButton -Content 'Doing it now' -Arguments 'ToastPS:now'
$LaterButton = New-BTButton -Content 'Not this time' -Arguments 'ToastPS:later'
```

Sorry all of that is rather disjointed, this last part with the protocol handlers is something I've only just started to wrap my head around and have yet to do proper documentation.

**Comment by timjdowning** [comment]
Hi Joshua. I really appreciate your response. I may yet give it a try. We're under a little bit of pressure at the moment to deal with all other working from home and return to work stuff.

Good luck with your work and development.

**Comment by Windos** [comment]
Hello, and thank you for your contribution to BurntToast.

Today marks the official release of [BurntToast v1.0.0](https://github.com/Windos/BurntToast/releases/tag/v1.0.0). As part of this major milestone and to ensure the project starts its next chapter with a clean slate, I am declaring "issue bankruptcy."

This decision is explained in more detail in the official announcement post: [Ten Years Toasting: BurntToast Hits the Big v1!](https://toastit.dev/2025/07/22/burnttoast-v1/)

As a result, I am closing all outstanding issues and pull requests (and discussions), including this one. Please do not interpret this as a lack of interest in the problem, suggestion, or topic you've raised. This is a necessary step to clear a very old backlog and focus development efforts on the current state of the module.

**If this topic is still relevant to you:**

1.  Please **open a new discussion**. You are welcome to link back to this one for historical context.

I hope this reset will allow me to be more responsive to community feedback going forward.

Thank you for your understanding and for being part of the BurntToast journey.
